File: csharp/LogicAndTrick.WikiCodeParser/Nodes/INode.cs
using System.Collections.Generic;

namespace LogicAndTrick.WikiCodeParser.Nodes
{
    public interface INode
    {
        string ToHtml();
        string ToPlainText();
        IList<INode> GetChildren();
        void ReplaceChild(int i, INode node);
        bool HasContent();
    }
}

File: php/src/Nodes/INode.php
<?php

namespace LogicAndTrick\WikiCodeParser\Nodes;

interface INode
{
    function ToHtml() : string;
    function ToPlainText() : string;
    /**
     * @return INode[]
     */
    function GetChildren() : array;
    function ReplaceChild(int $i, INode $node) : void;
    function HasContent() : bool;
}
